git-object/src/object/mod.rs

```diff
@@ -142,9 +162,26 @@ impl Object {
     }
 }
 
+use crate::decode::{loose_header, Error as DecodeError, LooseHeaderDecodeError};
 use crate::{BlobRef, CommitRef, Kind, ObjectRef, TagRef, TreeRef};
+use quick_error::quick_error;
+
+quick_error! {
+    #[derive(Debug)]
+    #[allow(missing_docs)]
+    pub enum LooseDecodeError {
+        InvalidHeader(err: LooseHeaderDecodeError) { from() }
+        InvalidContent(err: DecodeError) { from() }
+    }
+}
 
 impl<'a> ObjectRef<'a> {
+    /// Deserialize an object from a loose serialisation
+    pub fn from_loose(data: &'a [u8]) -> Result<ObjectRef<'a>, LooseDecodeError> {
+        let (kind, size, offset) = loose_header(data)?;
+        Ok(Self::from_bytes(kind, &data[offset..offset + size])?)
```

Comment:
Did you know that one can write `&data[offset..][..size])` here? So much easier on the eye and definitely optimized away :).

Reply:
Good point, I intellectually know about the pattern but I always have a hard time actually remembering to use it.

Reply:
It feels like I only learned it a month ago, but when I did it was added everywhere :D - so amazing.